cli/ttlog/logger.go

```diff
@@ -15,68 +22,113 @@ type LoggerOpts struct {
 	Prefix string
 }
 
-// Logger represents an active logging object.
-// A Logger can be used simultaneously from multiple goroutines;
-// it guarantees serialize access to the Writer.
-type Logger struct {
-	// Embedded logger, the functionality of which will be extended.
+type Logger interface {
+	// Logger API.
+	Printf(format string, v ...any)
+	Println(v ...any)
+	Fatal(v ...any)
+	Fatalf(format string, v ...any)
+	Writer() io.Writer
+
+	// Rotate re-opens a log file.
+	Rotate() error
+	// GetOpts returns the logger options that were used during creation.
+	GetOpts() LoggerOpts
+	// Close closes a log file.
+	Close() error
+}
+
+// writerLogger is a thin log.Logger wrapper providing the Logger interface.
+type writerLogger struct {
+	*log.Logger
+}
+
+// fileLogger is a logger that writes log messages to the file.
+type fileLogger struct {
```

Comment:
Maybe it will be better to restructure this file. So we would first we have `writerLogger` type and all its methods. And only then `fileLogger` and all its methods. Right now everything is mixed up.

I think this is a nit, up to you.

Reply:
Done.